src/print.c

```diff
@@ -245,6 +259,189 @@ launch_preview (const char *filename,
   return retval;
 }
 
+static int
+pdf_get_actions_page (char        *filename,
+                      PDF_ACTIONS  act,
+                      int          page)
+{
+  g_autoptr (GSubprocess) process = NULL;
+  GInputStream *stream;
+  g_autoptr(GPtrArray) args = g_ptr_array_new_full (8, g_free);
+  g_autoptr (GError) err = NULL;
+  char buffer[50] = { 0 };
+
+  g_ptr_array_add (args, g_strdup_printf ("%s", LIBEXECDIR "/xdg-desktop-portal-gtk-utils/pdftoraw"));
+  g_ptr_array_add (args, g_strdup_printf ("--file=%s", filename));
+  switch (act)
+  {
+    case PDF_TEST :
+       g_ptr_array_add (args, g_strdup ("--test"));
+       break;
+    case PDF_NUM_PAGES :
+       g_ptr_array_add (args, g_strdup ("--pages"));
+       break;
+    case PDF_WIDTH :
+       g_ptr_array_add (args, g_strdup_printf ("--width=%d", page));
+       break;
+    case PDF_HEIGHT :
+       g_ptr_array_add (args, g_strdup_printf ("--height=%d", page));
+       break;
+   default:
+       return 0;
+  }
+  g_ptr_array_add (args, NULL);
+
+  process = g_subprocess_newv ((const gchar * const *)args->pdata, G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL);
+
+  if (process)
+  {
+    stream = g_subprocess_get_stdout_pipe (process);
+    if (stream)
+    {
+      g_input_stream_read (stream, buffer, sizeof (buffer), NULL, NULL);
+      if (buffer[0])
+        return atoi (buffer);
+    }
+  }
+  return 0;
+}
+
+static unsigned char *
+pdf_get_data_page (char *filename,
+                   int   w,
+                   int   h,
+                   int   page)
+{
+  g_autoptr (GSubprocess) process = NULL;
+  GInputStream *stream;
+  unsigned char *data = NULL;
+  g_autoptr (GPtrArray) args = g_ptr_array_new_full (8, g_free);
+  int read = 0;
+  gsize total_read = 0;
+  g_autoptr (GError) err = NULL;
+
+  g_ptr_array_add (args, g_strdup_printf ("%s", LIBEXECDIR "/xdg-desktop-portal-gtk-utils/pdftoraw"));
+  g_ptr_array_add (args, g_strdup_printf ("--file=%s", filename));
+  g_ptr_array_add (args, g_strdup_printf ("--raw=%d", page));
+  g_ptr_array_add (args, NULL);
+  process = g_subprocess_newv ((const gchar * const *)args->pdata, G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL);
```

Comment:
This still needs to run under `bwrap` to be sandboxed.

Reply:
I don't understand, can you give me more details. Thank you

Reply:
This discussion: https://github.com/flatpak/xdg-desktop-portal-gtk/pull/366#issuecomment-999084174

Reply:
I think that bwrap is completely respected!
The executed process does not access the network, does not write to the disk and the return is done on STDOUT
If we talk about this line :
```
413  if ((fd2 = g_file_open_tmp (PACKAGE_NAME "XXXXXX", &filename, &err)) == -1)
414    return FALSE;
415
416  ostream = g_unix_output_stream_new (fd2, TRUE);
```

This is a copy of the code in the same file:
```
503  if ((fd2 = g_file_open_tmp (PACKAGE_NAME "XXXXXX", &filename, error)) == -1)
504    return FALSE;
505
506  ostream = (GUnixOutputStream *)g_unix_output_stream_new (fd2, TRUE);1
```
I am open to your suggestions.